src/AppInstallerCLICore/Commands/UninstallCommand.cpp

```diff
@@ -129,19 +130,29 @@ namespace AppInstaller::CLI
                 Workflow::GetManifestFromArg <<
                 Workflow::ReportManifestIdentity <<
                 Workflow::SearchSourceUsingManifest <<
-                Workflow::EnsureOneMatchFromSearchResult(true);
+                Workflow::EnsureOneMatchFromSearchResult(true) <<
+                Workflow::UninstallSinglePackage;
         }
         else
         {
-            // search for a single package to uninstall
-            context <<
-                Workflow::SearchSourceForSingle <<
-                Workflow::HandleSearchResultFailures <<
-                Workflow::EnsureOneMatchFromSearchResult(true) <<
-                Workflow::ReportPackageIdentity;
+            // search for specific packages to uninstall
+            context << Workflow::CheckForMultiQuery;
+            if (!context.Args.Contains(Execution::Args::Type::MultiQuery))
+            {
+                context <<
+                    Workflow::SearchSourceForSingle <<
+                    Workflow::HandleSearchResultFailures <<
+                    Workflow::EnsureOneMatchFromSearchResult(true) <<
+                    Workflow::ReportPackageIdentity <<
+                    Workflow::UninstallSinglePackage;
+            }
+            else
+            {
+                context <<
+                    Workflow::GetMultiSearchRequests <<
+                    Workflow::SearchSubContextsForSingle(Workflow::SearchSubContextsForSingle::SearchPurpose::Uninstall) <<
```

Comment:
After reading through this again, it seems like the implementation in #2877 of SearchResultType and the SearchPurpose here are similar; Would it make sense to align them both to `SearchPurpose` and use it more globally rather than inside the namespace of SearchSubContextsForSingle?

If so, I'll update that PR